source/common/router/upstream_request.cc

```diff
@@ -478,6 +493,15 @@ void UpstreamRequest::enableDataFromDownstreamForFlowControl() {
   }
 }
 
+void UpstreamRequest::overflowDataFromDownstream() {
+  ASSERT(parent_.upstreamRequests().size() == 1 || parent_.downstreamEndStream());
```

Comment:
Can you include some rationale for this ASSERT?

Reply:
I'll add more comments.
IIUC this is a special temporary buffer (buffer_request_body) that's being created for an upstream that wasn't yet initialized, and the data will later be moved to the upstream. See:
https://github.com/envoyproxy/envoy/blob/cfa29e187007331a93bfc83df5873aa7270b3980/source/common/router/upstream_request.cc#L180-L187